packages/eas-cli/src/commands/worker/deploy.ts

```diff
@@ -0,0 +1,195 @@
+        if (!json.success || !json.result || typeof json.result !== 'object') {
+          throw new Error(json.message ? `Upload failed: ${json.message}` : 'Upload failed!');
+        }
+        return json.result;
+      }
+    }
+
+    async function uploadAssetsAsync(
+      assetMap: WorkerAssets.AssetMap,
+      uploads: Record<string, string>
+    ): Promise<void> {
+      if (typeof uploads !== 'object' || !uploads) {
+        return;
+      }
+
+      // TODO(@kitten): Batch and upload multiple files in parallel
+      const uploadParams: UploadParams[] = [];
+      for await (const asset of WorkerAssets.listAssetMapFiles(distClientPath, assetMap)) {
+        const uploadURL = uploads[asset.normalizedPath];
+        if (uploadURL) {
+          uploadParams.push({ url: uploadURL, filePath: asset.path });
+        }
+      }
+
+      const progress = {
+        total: uploadParams.length,
+        pending: 0,
+        percent: 0,
+        transferred: 0,
+      };
+
+      const updateProgress = createProgressTracker({
+        total: progress.total,
+        message(ratio) {
+          const percent = `${Math.floor(ratio * 100)}`;
+          const details = chalk.dim(
+            `(${progress.pending} Pending, ${progress.transferred} Completed, ${progress.total} Total)`
+          );
+          return `Uploading client assets: ${percent.padStart(3)}% ${details}`;
+        },
+        completedMessage: 'Uploaded client assets for worker deployment',
+      });
+
+      try {
+        for await (const signal of batchUploadAsync(uploadParams)) {
+          if ('response' in signal) {
+            progress.pending--;
+            progress.percent = ++progress.transferred / progress.total;
+          } else {
+            progress.pending++;
+          }
+          updateProgress({ progress });
+        }
+      } catch (error: any) {
+        updateProgress({ isComplete: true, error });
+        Log.error(error);
+        return;
+      }
+      updateProgress({ isComplete: true });
+    }
+
+    let progress = ora('Preparing worker upload');
+    let assetMap: WorkerAssets.AssetMap;
+    let tarPath: string;
+    try {
+      assetMap = await WorkerAssets.createAssetMap(distClientPath);
+      tarPath = await WorkerAssets.packFilesIterable(emitWorkerTarballAsync(assetMap));
+    } catch (error: any) {
+      progress.fail('Failed to prepare worker upload');
+      Log.error(error);
+      return;
+    }
+    progress.succeed('Prepared worker upload');
+
+    progress = ora('Creating worker deployment');
+    let deployResult: any;
+    try {
+      deployResult = await uploadTarballAsync(tarPath);
+    } catch (error: any) {
+      progress.fail('Failed to create worker deployment');
+      Log.error(error);
+      return;
+    }
+    progress.succeed('Created worker deployment');
+
+    await uploadAssetsAsync(assetMap, deployResult.uploads);
+
+    const baseDomain = process.env.EXPO_STAGING ? 'staging.expo.app' : 'expo.app';
```

Comment:
does it make sense to also cover `expo.test` for `EXPO_LOCAL=1`?

Reply:
We can't deploy to `expo.test`, since it's Cloudflare-based. For that, we'd have to simulate worker namespaces, which isn't easily possible with Miniflare/Cloudflare's test tooling. The idea here is to either:
- just deploy to the Cloudflare staging environment anyway, even if your API is local
- or; to have a development environment in Cloudflare which we'll just use when running locally